### PR TITLE
feat(study): validate wbs on save

### PR DESCRIPTION
**Pull request by krsteen**

### CODE REVIEW

File: src/Sepes.Infrastructure/Service/Study/StudyWbsValidationService.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        public async Task ValidateForStudyCreateOrUpdate(Study study)
+        {
+            try
+            {
+                await PerformValidation(study, false);
+            }
+            catch (Exception ex)
+            {
```
Comment (HansKristianVikOlsen) [inline src/Sepes.Infrastructure/Service/Study/StudyWbsValidationService.cs:29]: Code scanning does not like generic catch clauses, should we disable this warning?

Comment (krsteen) [reply src/Sepes.Infrastructure/Service/Study/StudyWbsValidationService.cs:29]: I agree

File: src/Sepes.Tests/Services/Infrastructure/Study/StudyCreateServiceShould.cs [hunk 2]
```diff
@@ -35,7 +29,7 @@ public async Task CreatingStudyWithoutNameShouldFail(string name)
         [Theory]
         [InlineData(null)]
         [InlineData("")]
-        public async void CreatingStudyWithoutVendorShouldFail(string vendor)
+        public async void FailIfStudyIsMissingVendor(string vendor)
```
Comment (HansKristianVikOlsen) [inline src/Sepes.Tests/Services/Infrastructure/Study/StudyCreateServiceShould.cs:32]: Did you change naming convention or is this just so it is shorter?

**Review by HansKristianVikOlsen — APPROVED** [review]

File: src/Sepes.Tests/Services/Infrastructure/Study/StudyCreateServiceShould.cs (hunk 2 above)
Comment (krsteen) [reply src/Sepes.Tests/Services/Infrastructure/Study/StudyCreateServiceShould.cs:32]: Something i picked up at the TDD course, you see the class is named StudyCreateServiceShould, and then the Method name starts with Fail.., so the class name + test name reads as a sentence "StudyCreateServiceShould FailIfStudyIsMissingVendor"

Comment (HansKristianVikOlsen) [reply src/Sepes.Tests/Services/Infrastructure/Study/StudyCreateServiceShould.cs:32]: Ahh, that is clever!

**Pull request merged**